parser.js

```diff
@@ -346,10 +359,20 @@ function processReplay(file, opts = {}) {
       pdoc.region = pdata.m_toon.m_region;
       pdoc.realm = pdata.m_toon.m_realm;
 
+      pdoc.ToonHandle =
+        pdata.m_toon.m_region +
+        '-' +
+        pdata.m_toon.m_programId +
+        '-' +
+        pdata.m_toon.m_realm +
+        '-' +
+        pdata.m_toon.m_id;
+
       // ok so actually search forward here and look for a name match
       for (let j = i; j < data.tags.length; j++) {
-        if (data.tags[j].name === pdoc.name) {
+        if (pdoc.ToonHandle === data.tags[j].ToonHandle) {
           pdoc.tag = parseInt(data.tags[j].tag);
+          break;
         }
       }
```

Comment:
This code is the main reason for this PR; Changing this to using the `ToonHandle`, instead of name, ensures that the correct player is assigned the tag. 👍 
Everything else in this PR is to enable this bit of code.